clients/venice-push-job/src/main/resources/avro/ValidateSchemaAndBuildDictMapperOutput.avsc

```diff
@@ -0,0 +1,19 @@
+{
+  "name": "ValidateSchemaAndBuildDictMapperOutput",
+  "namespace": "com.linkedin.venice.hadoop.output.avro",
+  "type": "record",
+  "doc": "The data protocol between ValidateSchemaAndBuildDictMapper and VenicePushJob",
+  "fields": [
+    {
+      "name": "inputFileDataSize",
+      "doc": "Total input Data size in all files",
+      "type": "long"
+    },
+    {
+      "name": "zstdDictionary",
+      "doc": "Zstd compression dictionary",
+      "type": ["null", "bytes"],
```

Comment:
Do we need `null` as a union branch here? Can we just keep `bytes` and the value of the array could be `null`

Reply:
If we don't have `null` as a union branch here: default values for all the fields in the avro record (eg: `new byte[0] `for `zstdDictionary`) should be added explicitly. This works for now, but once we evolve this schema to add more fields, the code will start to look messy with multiple branches(or similar other mechanisms) to add the default values for all keys for which there are no valid data. Having `null` as union branch keeps it simple.

Reply:
What do you think on this?

Reply:
Sounds good